### PR TITLE
Enable test_action_communication to compile against available rmw

### PR DESCRIPTION
**Pull request by apojomovsky**
This PR adds enable the `test_action_communication` tests to compile and run against each rmw implementation.

### CODE REVIEW

File: rcl_action/CMakeLists.txt [hunk 1]
```diff
@@ -95,19 +96,70 @@ if(BUILD_TESTING)
...
+        file(TO_CMAKE_PATH "${RTI_BIN_PATH}" RTI_BIN_PATH)
+        set(TEST_PATH "${RTI_BIN_PATH};${ENV_PATH}")
+        if(NOT WIN32)
+          string(REPLACE ";" ":" TEST_PATH "${TEST_PATH}")
+        endif()
+      endif()
+    endif()
+
+    if(
+      rmw_implementation STREQUAL "rmw_connext_cpp" OR
```
Comment (jacobperron) [inline rcl_action/CMakeLists.txt]: Is there a reason why we only test for these specific implementations? What if we remove this condition?

Comment (apojomovsky) [reply rcl_action/CMakeLists.txt]: Oops, you're right, not needed at all! That was required for the test_security (from where I took this piece of cmake magic from).
Just updated it, thanks!

**Comment by sloretz** [comment]
Looks like the timeout of 10 seconds is too short for the connext tests

```
2: Running main() from gtest_main.cc
2: [==========] Running 13 tests from 1 test case.
2: [----------] Global test environment set-up.
2: [----------] 13 tests from TestActionCommunication__rmw_connext_cpp
2: [ RUN      ] TestActionCommunication__rmw_connext_cpp.test_valid_goal_comm
2: [       OK ] TestActionCommunication__rmw_connext_cpp.test_valid_goal_comm (3311 ms)
2: [ RUN      ] TestActionCommunication__rmw_connext_cpp.test_valid_cancel_comm
2: [       OK ] TestActionCommunication__rmw_connext_cpp.test_valid_cancel_comm (3191 ms)
2: [ RUN      ] TestActionCommunication__rmw_connext_cpp.test_valid_result_comm
 2/14 Test  #2: test_action_communication___rmw_connext_cpp ....***Timeout  10.09 sec
```

**Comment by apojomovsky** [comment]
Thanks for the suggestion @sloretz , I just updated the timeout and re-triggered CI.

CI status:
Linux: [Build Status](https://ci.ros2.org/job/ci_linux/5807/)
Linux aarch64: [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/2374/)
MacOS: [Build Status](https://ci.ros2.org/job/ci_osx/4801/)
Windows: [Build Status](https://ci.ros2.org/job/ci_windows/5705/)

**Comment by apojomovsky** [comment]
Green light on CI so I'm merging this one.

**Pull request merged**